src/test/java/com/github/commonsrdf/dummyimpl/GraphImpl.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.commonsrdf.dummyimpl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.github.commonsrdf.api.BlankNodeOrIRI;
+import com.github.commonsrdf.api.Graph;
+import com.github.commonsrdf.api.IRI;
+import com.github.commonsrdf.api.RDFTerm;
+import com.github.commonsrdf.api.Triple;
+
+public class GraphImpl implements Graph {
+
+	private static final int TO_STRING_MAX = 10;
+	protected List<Triple> triples = new ArrayList<Triple>();
```

Comment:
Although there may be various implementation differences related to SPARQL-1.1 and duplicates from queries, at the abstract RDF-1.1 level Graph is defined to be a set so we shouldn't need to keep duplicates in a reference implementation.

http://www.w3.org/TR/rdf11-concepts/#data-model

I don't think there is any reason for the abstract test to be testing afterCount > beforeCount for duplicates. I missed that bit when I went through. If you know you are working with a List backed Graph then you could add that test in for that implementation to verify its specific implementation.